vhdbuilder/packer/import-image.sh

```diff
@@ -30,3 +44,37 @@ azcopy-preview copy $IMPORT_IMAGE_URL$IMPORT_IMAGE_SAS $DESTINATION_WITH_SAS
 cp $SETTINGS_JSON $SETTINGS_JSON.tmp
 jq --arg imgurl "$IMPORTED_IMAGE_URL" '.imported_image_url = $imgurl' $SETTINGS_JSON.tmp > $SETTINGS_JSON
 rm $SETTINGS_JSON.tmp
+
+if [[ $HYPERV_GENERATION == "V2" ]]; then
+    az image create \
+        --resource-group $AZURE_RESOURCE_GROUP_NAME \
+        --name $IMPORT_NAME \
+        --source $IMPORTED_IMAGE_URL \
+        --hyper-v-generation V2 \
+        --os-type Linux
+
+    az sig image-definition create \
+        --resource-group $AZURE_RESOURCE_GROUP_NAME \
+        --gallery-name $SIG_GALLERY_NAME \
+        --gallery-image-definition $IMPORT_NAME \
+        --location $AZURE_LOCATION \
+        --os-type Linux \
+        --publisher microsoft-aks \
+        --offer $IMPORT_NAME \
+        --sku $OS_SKU \
+        --hyper-v-generation V2 \
+        --os-state generalized \
+        --description "Imported image for AKS Packer build"
+
+    az sig image-version create \
+        --location $AZURE_LOCATION \
+        --resource-group $AZURE_RESOURCE_GROUP_NAME \
+        --gallery-name $SIG_GALLERY_NAME \
+        --gallery-image-definition $IMPORT_NAME \
+        --gallery-image-version 1.0.0 \
+        --managed-image $IMPORT_NAME
+
+    cp $SETTINGS_JSON $SETTINGS_JSON.tmp
+    jq --arg imgname "$IMPORT_NAME" '.imported_sig_image = $imgname' $SETTINGS_JSON.tmp > $SETTINGS_JSON
+    rm $SETTINGS_JSON.tmp
+fi
```

Comment:
I feel it's better to move this importing step to be called from init-variable.sh
1. logically this is an initialization step, "build-packer" should be able to just build
2. having to update the setting.json indicates the workflow is not the optimal.